neonvm-runner/cmd/main.go

```diff
@@ -1104,6 +1111,28 @@ func runQEMU(
 	return err
 }
 
+func getQemuBinaryName(architecture string) string {
+	switch architecture {
+	case architectureArm64:
+		return qemuBinArm64
+	case architectureAmd64:
+		return qemuBinX8664
+	default:
+		panic(fmt.Errorf("unknown architecture %s", architecture))
+	}
+}
+
+func getMachineType(architecture string) string {
```

Comment:
It'd be good to have a comment explaining the situation here -- why these particular machine types?

Reply:
tbh there are no particular reason. for q35 it is probably historical reason and virt is the most generic arm platform which we use in runners already.

Reply:
perhaps @cicdteam can add some context on why q35?

Reply:
Sorry, I just saw this discussion.

In the x86_64 architecture, QEMU has two groups of emulated machines - `pc-q35` and `pc-i440fx` (yes, there are a couple of others, like `microvm` or `none`, but this is not our case). The `q35` series is more recent, so it was chosen. In fact, there are no other options for x86_64. 

FYI: `-machine q35` is alias of `-machine pc-q35-9.0` (in QEMU v9.0.1),by other words - most recent hardware.

Reply:
added some clarifications on that